lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -706,16 +710,20 @@ def cmd_services(*args)
             # Sentinel value meaning all
             host_ranges.push(nil) if host_ranges.empty?
             ports = nil if ports.empty?
+            matched_service_ids = []
 
             each_host_range_chunk(host_ranges) do |host_search|
-              framework.db.services(framework.db.workspace, onlyup, proto, host_search, ports, names).each do |service|
+              break if !host_search.nil? && host_search.empty?
+              framework.db.services(framework.db.workspace, onlyup, nil, host_search, nil, nil, search_term).each do |service|
```

Comment:
Why is `nil` being passed in for the `proto`, `ports` and `names` parameters?

Reply:
The old functionality was that you could search on port or protocol by passing in the associated flags. This was inconsistent with other commands like `hosts` or `loot` where you only search based on the search term passed with `-S`. I changed it to be consistent with the other commands. 

I also didn't want to change the function signature at this time, so I still have to pass something in for those other values.

Reply:
Why didn't you want to change the signature at this time?

Reply:
There are a ton of calls to the method and I think it's scope creep to tackle that at this time. We also haven't done it for other models. We should probably just create a backlog ticket and handle them all at once.

Reply:
+1 on the backlog item